### DISCUSSION TITLE
About wasm integration

### DISCUSSION THREAD

**Discussion by geoman2** [luisaveiro/playground#4] [Ideas]
Hey Luis, have you thought about running all these gems you grabbed here as wasm modules? There is a wasm server from vmware with support for python, php, rust, ruby and node runtimes with rust based webserver actix web running as the router (to handle request and response, etc), see https://wasmlabs.dev/articles/wasm-workers-server-1-0-0/

What do you think, will give it a try?

**Comment by geoman2** [comment]
There is an article on their blog running a php script in 4 scenarios with explanations - php native, php docker, php wasm and php wasm inside docker and also a video in the link above running two different wasm runtimes as part of the same app.
From what I've seen, the images wasm runtime uses, are based on boolseye slim debian image, something weighting 27 MB. Would you consider migrating all or a part of the apps also to this image, are there impediments not to migrate, as I've seen you used various images for different apps?

**Comment by luisaveiro** [comment]
@geoman2 thank you for your suggestion. I'm not familiar with wasm. Thank you for sharing the article. I'll have a read-through. At the moment, I'm not planning on adding support for wasm. However, I will keep wasm in mind in the future.
